src/main/java/org/folio/circulation/resources/SlipsResource.java

```diff
@@ -117,9 +118,26 @@ private void getMany(RoutingContext routingContext) {
     final var servicePointRepository = new ServicePointRepository(clients);
     final var patronGroupRepository = new PatronGroupRepository(clients);
     final var departmentRepository = new DepartmentRepository(clients);
+    final var configurationRepository = new ConfigurationRepository(clients);
     final UUID servicePointId = UUID.fromString(
       routingContext.request().getParam(SERVICE_POINT_ID_PARAM));
 
+    if ("searchSlips".equals(collectionName) && requestType == RequestType.HOLD) {
+      configurationRepository.lookupPrintHoldRequestsEnabled()
+        .thenAccept(result -> result.next(printHoldRequestsConfiguration -> {
+          if (printHoldRequestsConfiguration == null ||
+            !printHoldRequestsConfiguration.isPrintHoldRequestsEnabled()) {
+            log.info("getMany:: Print hold requests configuration is disabled");
+            context.writeResultToHttpResponse(succeeded(JsonHttpResponse.ok(
+               new io.vertx.core.json.JsonObject()
+                  .put("searchSlips", new io.vertx.core.json.JsonArray())
+                  .put("totalRecords", 0)
+            )));
+          }
+          return null;
+        }));
```

Comment:
move this lambda to a separate method, something like 
 returnNoRecordsIfSearchSlipsDisabled(WebContext context)